dotnet/samples/Demos/OnnxSimpleRAG/Program.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable SKEXP0070
+#pragma warning disable SKEXP0050
+#pragma warning disable SKEXP0001
+
+using System;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Plugins.Memory;
+
+// Ensure you follow the preparation steps provided in the README.md
+
+// Path to the folder of your downloaded ONNX PHI-3 model
+var chatModelPath = @"C:\path\to\huggingface\Phi-3-mini-4k-instruct-onnx\cpu_and_mobile\cpu-int4-rtn-block-32";
+
+// Path to the file of your downloaded ONNX BGE-MICRO-V2 model
+var embeddingModelPath = @"C:\path\to\huggingface\bge-micro-v2\onnx\model.onnx";
+
+// Path to the vocab file your ONNX BGE-MICRO-V2 model
+var embeddingVocabPath = @"C:\path\to\huggingface\bge-micro-v2\vocab.txt";
+
+// Load the services
+var builder = Kernel.CreateBuilder()
+    .AddOnnxRuntimeGenAIChatCompletion("phi-3", chatModelPath)
+    .AddBertOnnxTextEmbeddingGeneration(embeddingModelPath, embeddingVocabPath);
+
+// Build Kernel
+var kernel = builder.Build();
+
+// Get the instances of the services
+var chatService = kernel.GetRequiredService<IChatCompletionService>();
+var embeddingService = kernel.GetRequiredService<ITextEmbeddingGenerationService>();
+
+// Create a memory store and a semantic text memory
+var memoryStore = new VolatileMemoryStore();
+var semanticTextMemory = new SemanticTextMemory(memoryStore, embeddingService);
```

Comment:
Use the new memory abstractions

Reply:
Currently there's no substitute for that. I created the issue below to track it:

- #8542